pkgs/top-level/stage.nix

```diff
@@ -97,7 +97,8 @@ let
       res self;
     in res;
 
-  aliases = self: super: import ./aliases.nix super;
+  aliases = self: super: if config.skipAliases or false then {}
```

Comment:
Yes, we could even put a warning in aliases as well. It would probably be too annoying right now though- aliases are still used extensively in Nixpkgs.